src/components/common/Interactive.tsx

```diff
@@ -40,21 +43,31 @@ interface Props {
 
 const InteractiveBase = ({ onMove, onKey, ...rest }: Props) => {
   const container = useRef<HTMLDivElement>(null);
-  const hasTouched = useRef(false);
-  const [isDragging, setDragging] = useState(false);
   const onMoveCallback = useEventCallback<Interaction>(onMove);
   const onKeyCallback = useEventCallback<Interaction>(onKey);
+  const hasTouch = useRef(false);
 
-  // Prevent mobile browsers from handling mouse events (conflicting with touch ones).
-  // If we detected a touch interaction before, we prefer reacting to touch events only.
-  const isValid = (event: MouseEvent | TouchEvent): boolean => {
-    if (hasTouched.current && !isTouch(event)) return false;
-    if (!hasTouched.current) hasTouched.current = isTouch(event);
-    return true;
-  };
+  const [handleMoveStart, handleKeyDown] = useMemo(() => {
```

Comment:
Common `useMemo` prevents us from `useCallback` hell and it makes code easier to read, as all logic in one place